src/ray/raylet/node_manager.cc

```diff
@@ -1400,32 +1402,48 @@ void NodeManager::ProcessDisconnectClientMessage(
   // Publish the worker failure.
   auto worker_failure_data_ptr =
       gcs::CreateWorkerFailureData(self_node_id_, worker->WorkerId(), worker->IpAddress(),
-                                   worker->Port(), time(nullptr), intentional_disconnect);
+                                   worker->Port(), time(nullptr), disconnect_type);
   RAY_CHECK_OK(
       gcs_client_->Workers().AsyncReportWorkerFailure(worker_failure_data_ptr, nullptr));
 
   if (is_worker) {
     const ActorID &actor_id = worker->GetActorId();
     const TaskID &task_id = worker->GetAssignedTaskId();
-    // If the worker was running a task or actor, clean up the task and push an
-    // error to the driver, unless the worker is already dead.
+
     if ((!task_id.IsNil() || !actor_id.IsNil()) && !worker->IsDead()) {
+      // If the worker was running a task or actor, clean up the task and push an
+      // error to the driver, unless the worker is already dead.
+      // TODO(rkn): Define this constant somewhere else.
+      std::string type_str;
+      std::ostringstream error_message;
+
+      bool intentional_exit = false;
+      switch (disconnect_type) {
+      case rpc::ClientDisconnectType::UNEXPECTED_EXIT:
+        type_str = "worker_died";
+        error_message << "A worker died or was killed while executing task " << task_id
+                      << ".";
+        break;
+      case rpc::ClientDisconnectType::PLACEMENT_GROUP_REMOVED:
+        type_str = "placement_group";
+        error_message << "A worker was killed while executing task " << task_id
+                      << " due to placement group removal."
+                      << ".";
+        break;
+      default:
```

Comment:
case rpc::ClientDisconnectType::FINISHED:
case rpc::ClientDisconnectType:: UNUSED_RESOURCE_RELEASED:
  intentional_exit = true;

Reply:
Or you can add a comment to explain what default is.